docs/2-Interviews/answering_technical_questions.md

```diff
@@ -1,39 +1,129 @@
+
+### Algorithms: starting the interview
+
+- State what language you intend to use. Many C-style (curly brace) languages look similar but have subtle difference
+- Get started right away (on something)
+  - Don't think to yourself, or go back and forth on things for too long. It's perfectly fine to take a few minutes to time to think in the beginning, but interviewers want to see things moving at some point
+  - You don't have to get *coding* right away. In fact, you probably shouldn't
+  - Some things you can start with is to ask questions, draw some examples, describe possible approaches, write pseudocode
+- Ask questions!!!
+  - Clarify the problem to make sure you know what you're being asked (explain it back)
+  - Clarify requirements and assumptions (ask about edge cases, or clarifying details). E.g. shuffle a deck of cards - do I know how many cards? what does shuffling mean in this case?
+  - Go over 1-2 small test cases to make sure you know how the algorithm will work
+  - Note: figuring out what to ask is something you learn over time. Typical questions include: "what is the input size?", "is the input sorted?", "can I assume the input is always valid/well-formatted?", "how much memory do I have?".
+- Explain to the interviewer how you intend to approach and solve the problem at a conceptual level before you start writing code
+  - It helps you clarify your own thoughts
+  - It helps the interviewer follow along when you start coding
+  - It allows the interviewer to correct you if you misunderstood the question
+  - It allows the interviewer to jump in if the solution is totally off-track. Better to know early on before you write too much code going down the wrong path!
+  - Write stuff down as you think so they can see that you can organize your thoughts
+- It's okay to start with an inefficient approach, even brute force! Just get started, you can optimize things later - better to get a simple solution than no solution. Just don't forget to tell the interviewer that you're going for the brute force route. You don't want the interviewer to think that you think O(c^n) solution is efficient. So just tell the interviewer that you are aware that this is not the most efficient solution, but that writing even an inefficient approach will give you insight into the problem
+
+### Algorithms: solving the problem
+
+- Explain your thought process. Keep talking throughout the interview. Humans (interviewers included) tend to be uncomfortable with long silences for whatever reason.
```

Comment:
The interviewer also wants to know how you think, and what's going on in your mind throughout the interview (it's okay to verbalize feelings of nervousness too!)